### ISSUE TITLE
gmail wrongly rejects mailing list posts as spam

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#3555]
I allowed 3 messages within an hour and all gmail addresses started bouncing the third one with:
```
<USER@gmail.com>](mailto:USER@gmail.com): host gmail-smtp-in.l.google.com[108.177.15.26] said:
    550-5.7.1 [78.129.163.13      12] Our system has detected that this message
    is 550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to
    Gmail, 550-5.7.1 this message has been blocked. Please visit 550-5.7.1
    https://support.google.com/mail/?p=UnsolicitedMessageError 550 5.7.1  for
    more information. p8-20020a5d59a8000000b0021849827fccsi5831542wrr.45 -
    gsmtp (in reply to end of DATA command)
```
We have hundreds of gmail addresses on the mailing list... so hundreds of bounces.

This is where it gets "fun", the chat support was not very encouraging, asking questions that had obvious answers in the log samples already provided or telling me that "_I understand that you are concerned about spam emails_" when I am not concerned about that at all, only about gmail mishandling emails:
![image](https://user-images.githubusercontent.com/1049737/174249802-8a341f0b-41b0-4a7f-a9f1-4da223b76482.png)

Then a fairly long (10 messages) email thread started regarding this issue, where I had to repeat the same thing again, and again.
Here's a sample exchange:
> > This is about the issue for not being able to send emails through your hosted mailing list.
> > As per our higher tier of support, you need to get in touch with your email provider so they can check their server about the said error. 
> > May I ask if you already tried to reach out to them about this case?
>
> No, I do not need to "get in touch with my email provider".
> I am my own email provider and I also run the mailing list on that same system.
> I do not need to check the server "about the said error" as I've already given you all the details about the error, which comes from gmail's server wrongly rejecting the mailing list messages as spam.
>
> Please fix gmail's spam filter for this mailing list so that I can resume traffic and re-send the message that has been wrongly blocked.

They also tried to "inform" me about the error code, which *I* had given to them.
On two more occasions, they asked again that I talked to my "email provider". Either this was a different person who had not read the earlier discussion, or just forgotten about it?

Eventually this response came:
> The main issue was on Gmail server and our Higher Tier Team doesn't have tools to troubleshoot on this kind of delivery issue since we don't have access to Gmail servers.

Thanks for wasting my time and resources google!

TLDR: don't use gmail to reliably receive mailing list messages.

Edit: It's not like they don't know about the problem either:
[Check if your Gmail message is authenticated
](https://support.google.com/mail/answer/180707) : _Important: Messages that aren't authenticated aren't necessarily spam. Sometimes authentication doesn't work for real organizations who send mail to big groups, like messages sent to mailing lists._

**Comment by totaam** [comment]
And again today for this message:
[ANNOUNCE xpra-html5 releases: v5.3 and v8.1](https://lists.devloop.org.uk/pipermail/shifter-users/2023-July/002984.html)
```
[<********>](mailto:********): host aspmx.l.google.com[66.102.1.27] said:
    550-5.7.1 [78.129.163.13      12] Our system has detected that this message
    is 550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to
    Gmail, 550-5.7.1 this message has been blocked. Please visit 550-5.7.1
    https://support.google.com/mail/?p=UnsolicitedMessageError 550 5.7.1  for
    more information. w6-20020a5d6086000000b0031134a78ed6si1388140wrt.155 -
    gsmtp (in reply to end of DATA command)
```

**Comment by totaam** [comment]
![Screenshot from 2023-07-18 22-49-00](https://github.com/Xpra-org/xpra/assets/1049737/9ae91538-ea27-44e1-991c-f53dcfa76afd)

**Comment by totaam** [comment]
So, they've changed their policy to require DMARC + DKIM. That was all.

The latest mailing list post made it:
https://lists.devloop.org.uk/pipermail/shifter-users/2023-July/002986.html

**Comment by totaam** [comment]
I sent two messages in the same afternoon... and it's triggering gmail again. Because of course it does, `DMARC` and `DKIM` notwithstanding.
```
[<REDACTED@REDACTED.com>](mailto:REDACTED@REDACTED.com): host aspmx.l.google.com[64.233.166.26] said:
    550-5.7.1 [78.129.163.13      19] Gmail has detected that this message is
    likely 550-5.7.1 suspicious due to the very low reputation of the sending
    domain. To 550-5.7.1 best protect our users from spam, the message has been
    blocked. 550-5.7.1 Please visit 550 5.7.1
    https://support.google.com/mail/answer/188131 for more information.
    q4-20020adfea04000000b003316947d2acsi6291692wrm.250 - gsmtp (in reply to
    end of DATA command)
```

I was going to try to continue using the mailing list for a little while longer but this may the final nail in the coffin.
I just don't have time to deal with this nonsense, again and again.

**Comment by totaam** [comment]
Amusingly, the latest release announcement did make it to gmail this time, but landed in the spam box, together with an email sent from the GSOC mailing list.
Google's own mailing list was also classified as spam!

**Comment by totaam** [comment]
And a new one today: `Your email has been rate limited because SPF authentication didn't 421-4.7.27 pass for this message. Gmail requires all bulk email senders to 421-4.7.27 authenticate with SPF.`
Which is blatantly untrue, SPF has been there for years and years.
The message did get through, just slower than usual.

**Comment by totaam** [comment]
[Google banned our GSoC Mailing List](https://groups.google.com/g/google-summer-of-code-mentors-list/c/Myb4_zBJWLw/m/sA_zA8V7AQAJ)

**Comment by totaam** [comment]
[Meta blocked Distrowatch links on Facebook while running Linux servers](https://www.theregister.com/2025/01/28/facebook_blocks_distrowatch/) - "AI"

**Comment by totaam** [comment]
New one today for the 6.2.4 release:
```
mail postfix/smtp[15243]: 79A3ACE836: host gmail-smtp-in.l.google.com[74.125.206.26] said:
    421-4.7.29 Your email has been rate limited because this message wasn't sent 421-4.7.29 over a TLS connection. Gmail requires all bulk email senders to use 421-4.7.29 TLS/SSL for SMTP connections.
    To set up TLS for email, visit 421-4.7.29  https://support.google.com/a?p=tls-ssl
    421-4.7.29 To learn more about Gmail requirements for bulk senders, visit 421 4.7.29
    https://support.google.com/a?p=sender-guidelines. ffacd0b85a97d-390e47aafc6si10583781f8f.246 - gsmtp (in reply to end of DATA command)
```

**Comment by totaam** [comment]
Good read: [Cory Doctorow's pluralistic 2025/01/23](https://pluralistic.net/2025/01/23/defense-in-depth/#self-marginalization): _That said, decentralization and federation are not perfect, set-and-forget defenses. Take email – the oldest, most successful federated system of them all. Email is nominally decentralized, but most email traffic goes through a handful of extremely large servers run by a cartel of companies (Google, Apple, Microsoft, and a few ISPs). These companies collude (or, more charitably, coordinate) to block email from non-cartel companies, in the name of fighting spam. This makes running your own mail server so hard that it is nearly impossible (that is, if you care about people actually receiving the email you send them):_
[Dead letters](https://pluralistic.net/2021/10/10/dead-letters/)

**Comment by totaam** [comment]
[Bill Cole on mastodon](https://toad.social/@grumpybozo/114213600922816869): _The stats we collect for the #SpamAssassin project (mass-scan results from participating sites) have long shown that spammers are more consistent at making SPF, DKIM, and DMARC correct than are legitimate senders. DMARC in particular has no discernible benefit for most senders, so it is a useless signal.
Rejecting mail based solely on authentication failures of those deeply flawed authentication methods does more harm than good._
